doc/bum_storm_control/bum_storm_control_hld.md

```diff
@@ -0,0 +1,398 @@
+3) create_policer SAI API is invoked to create a policer with the given input parameters. 
+4) The identifier of the policer created is associated with the encoded policer_name.
+5) The policer identifier is passed to set_port_attribute SAI API to set the appropriate type of storm-control on the port. 
+
+
+## 3.2 DB Changes
+This section describes the changes made to different DBs for supporting storm-control.
+
+
+### 3.2.1 CONFIG_DB
+A new table PORT_STORM_CONTROL is introduced in the configuration database for the purpose of storing storm-control configuration parameters. This table is filled by the management framework. 
+#### PORT_STORM_CONTROL
+
+    ;Store Storm Control configuration per physical port
+    ;Status: work in progress
+    ;storm control type - broadcast / unknown-unicast / unknown-multicast
+    key     = PORT_STORM_CONTROL:port|storm_control_type ; Ethernet Interface Name and storm control type
+    ;field  = value
+    enabled = BIT          ; Is the storm control enabled (1) or disabled (0) on the interface 
+    kbps     = 1*13 DIGIT   ; CIR value in kilo bits per second
+         
+### 3.2.2 APP_DB
+No tables are introduced in APP_DB
+### 3.2.3 STATE_DB
+No tables are introduced in STATE_DB. 
+
+### 3.2.4 ASIC_DB
+No changes are introduced in ASIC_DB. 
+### 3.2.5 COUNTER_DB
+No changes are introduced in COUNTER_DB. 
+
+
+## 3.3 Switch State Service Design
+### 3.3.1 PolicerOrch changes 
+Policer Orchestration agent is responsible for the following activities:
+   - Subscribes to notifications on PORT_STORM_CONTROL entries in the CONFIG_DB. 
+   - Creates an internal name for policer by encoding the interface name and storm-control type. 
+   - Creates the policer based on the parameters and associates the policer identifier to internal name
+   - Populates the port attribute SAI structures and pushes the entry to ASIC_DB. 
+
+
+## 3.4 SAI
+Storm-control SAI interface APIs are already defined. The table below represents the SAI attributes which shall be used for BUM storm-control.
+
+
+###### Table 2: Storm Control SAI attributes
+| Storm Control component  | SAI attributes                                         |                          
+|--------------------------|--------------------------------------------------------|
+| Meter type               | SAI_POLICER_ATTR_METER_TYPE                            |
+| Meter mode               | SAI_POLICER_ATTR_MODE                                  |
+| CIR (bps)                | SAI_POLICER_ATTR_CIR                                   |
+| Unknown-unicast policer  | SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID           |
+| Broadcast policer        | SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID       |
+| Unknown-Multicast policer| SAI_PORT_ATTR_MULTICAST_STORM_CONTROL_POLICER_ID       |
```

Comment:
Hi
This policer types doesn't seem accurate...
According to SAI attribute definition,
- `SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID` represents unknown unicast and multicast instead of unknown unicast
- `SAI_PORT_ATTR_MULTICAST_STORM_CONTROL_POLICER_ID` represents known multicast instead of unknown multicast.

Should we adjust the CLI according to the SAI attribute definition?